File: app/src/main/java/com/kakacat/minitool/epidemicinquiry/Contract.java
package com.kakacat.minitool.epidemicinquiry;

import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;
import com.kakacat.minitool.epidemicinquiry.model.GroupBean;

import java.util.List;

public interface Contract {

    interface Presenter extends IPresenter {
        void requestData();

        List<GroupBean> getGroupList();
    }

    interface View extends IView<Presenter> {
        void onUpdateViewSuccessful();

        void onUpdateViewError(String error);
    }
}

File: app/src/main/java/com/kakacat/minitool/epidemicinquiry/Contract.kt
package com.kakacat.minitool.epidemicinquiry

import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView
import com.kakacat.minitool.epidemicinquiry.model.GroupBean

interface Contract {
    interface Presenter : IPresenter {
        fun requestData()
        val groupList: MutableList<GroupBean>?
    }

    interface View : IView<Presenter> {
        fun onUpdateViewSuccessful()
        fun onUpdateViewError(error: String)
    }
}